### PR TITLE
improvement(inputconnected.tsx): added toggle dispatch on inputconnected

### PR DESCRIPTION
**Pull request by FelixBlaisThon**
### Proposed Changes

Added a isdirty toggle dispatch onChange for inputConnected so that the input gets dirty if not empty and displays error message if empty

### Acceptance Criteria

-   [x] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Comment by dmgauthier** [comment]
> I'm not sure about the change here, can you explain why you want the input to not be dirty when the value is invalid?
> 
> I think it needs to be dirty as soon as the value changes, even if it's an invalid value

I maybe missed a conversation about it, but I tend to agree with that :point_up:

File: packages/react-vapor/src/components/input/tests/InputConnected.spec.tsx [hunk 1]
```diff
@@ -172,5 +175,18 @@ describe('<InputConnected />', () => {
             expect(validate(newInputState.value)).toBe(false);
             expect(newInputState.valid).toBe(validate(newInputState.value));
         });
+        it('should call changeToDirty is set as props', () => {
+            const changeToDirtySpy = jasmine.createSpy();
+            const wrapperInputConnected = shallowWithState(<InputConnected />, {});
```
Comment (pochretien) [inline packages/react-vapor/src/components/input/tests/InputConnected.spec.tsx]: I think you can use the .dive() on shallowWithState
```
wrapperInputConnected.find(input).simulate('onchange');
```
and validate if the props is called with the full loop with the input element

File: packages/react-vapor/src/components/input/tests/InputConnected.spec.tsx [hunk 2]
```diff
@@ -20,8 +21,10 @@ describe('<InputConnected />', () => {
     });
 
     afterEach(() => {
-        store.dispatch(clearState());
-        wrapper.detach();
+        if (wrapper) {
+            store.dispatch(clearState());
+            wrapper.detach();
+        }
```
Comment (gdostie) [inline packages/react-vapor/src/components/input/tests/InputConnected.spec.tsx]: We probably want to clear the state even if we don't mount during the test?

```ts
store.dispatch(clearState());
wrapper?.detach();
```

**Comment by gdostie** [comment]
There is a failing unit test that would need to be fixed, other than that it looks good.

**Pull request merged**